### PR TITLE
Update rustls 0.20

### PR DESCRIPTION
**Pull request by quininer**

### CODE REVIEW

File: transports/websocket/src/tls.rs [hunk 1]
```diff
@@ -105,35 +116,39 @@ impl Builder {
...
             .map_err(|e| Error::Tls(Box::new(e)))?;
         Ok(self)
     }
 
     /// Finish configuration.
     pub fn finish(self) -> Config {
+        let client = rustls::ClientConfig::builder()
+            .with_safe_defaults()
+            .with_root_certificates(self.client_root_store)
+            .with_no_client_auth();
```
Comment (thomaseizinger) [inline transports/websocket/src/tls.rs:141]: I am not 100% what should be happening here but does this make sense if we are constructing a _client_ `Config` here?

File: transports/websocket/src/tls.rs [hunk 2]
```diff
@@ -69,33 +70,43 @@ impl Config {
 
     /// Create a client-only configuration.
     pub fn client() -> Self {
+        let client = rustls::ClientConfig::builder()
+            .with_safe_defaults()
+            .with_root_certificates(client_root_store())
+            .with_no_client_auth();
```
Comment (thomaseizinger) [inline transports/websocket/src/tls.rs:76]: Same here.

**Review by thomaseizinger — CHANGES_REQUESTED** [review]
One comment, otherwise LGTM!

Can you please address the clippy lint failure?

File: transports/websocket/src/tls.rs (hunk 1 above)
Comment (quininer) [reply transports/websocket/src/tls.rs:141]: I'm not sure, I just translated the old code as is.

Comment (thomaseizinger) [reply transports/websocket/src/tls.rs:141]: The old code only sets `NoClientAuth` within the `server` constructor (line 108).

For `client`, we don't set anything (line 89).

Comment (quininer) [reply transports/websocket/src/tls.rs:141]: `NoClientAuth` is the default value in rustls 0.19, but in rustls 0.20 you need to set it explicitly.

Comment (thomaseizinger) [reply transports/websocket/src/tls.rs:141]: Right, thanks for clarifying!

**Comment by mxinden** [comment]
Replaced with https://github.com/libp2p/rust-libp2p/pull/2336. Thanks @quininer for the help!

**Pull request closed without merging**